internal/testutils/helpers.go

```diff
@@ -15,14 +16,36 @@ func SkipIfNotRunningIntegrationTests(t *testing.T) {
 	}
 }
 
-func GetSourcesCountWithAppType(appTypeId int64) int {
-	var sourcesCount int
+func GetSourcesWithAppType(appTypeId int64) []model.Source {
+	var sourceIdList []int64
 
+	// Find applications with given application type and create
+	// list of unique source IDs
 	for _, app := range fixtures.TestApplicationData {
 		if app.ApplicationTypeID == appTypeId {
-			sourcesCount++
+			var idExists bool
+			for _, id := range sourceIdList {
+				if app.SourceID == id {
+					idExists = true
+					break
+				}
+			}
+			if !idExists {
+				sourceIdList = append(sourceIdList, app.SourceID)
+			}
 		}
 	}
 
-	return sourcesCount
+	// Find sources for source IDs
+	var out []model.Source
+	for _, id := range sourceIdList {
+		for _, src := range fixtures.TestSourceData {
+			if id == src.ID {
+				out = append(out, src)
+				break
+			}
+		}
+	}
+
+	return out
 }
```

Comment:
That's good! I'm going to provide an alternative implementation too using maps which might be slightly more concise. The idea is for you to get comfy around maps too :)

```go
func GetSourcesWithAppType(appTypeId int64) []model.Source {
	// We are only interested in the keys, so we use a "struct{}" as a value so that Go doesn't allocate memory for it.
	var sourceIds = make(map[int64]struct{})
	
	// Find applications with given application type and create
	// list of unique source IDs
	for _, app := range fixtures.TestApplicationData {
		if app.ApplicationTypeID == appTypeId {
			_, ok := sourceIds[app.SourceID]
			if !ok {
				sourceIds[app.SourceID] = struct{}{}
			}
		}
	}

	var sources []model.Source
	for id := range sourceIds {
		for _, src := range fixtures.TestSourceData {
			if id == src.ID {
				sources = append(sources, src)
			}
		}
	}

	return sources
}
```

Reply:
thanks @MikelAlejoBR this is really better